tesseract_monitoring/src/robot_comm_monitor_node.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * robot_communication_monitor.cpp
+ *
+ * Checks for fanuc communication errors
+ *
+ * Created on June 14, 2019
+ *    Author: rwall
+ *
+ *
+ */
+
+#include "boost/bind.hpp"
+#include "ros/ros.h"
+#include "std_msgs/Bool.h"
+#include "std_msgs/String.h"
+#include <sensor_msgs/JointState.h>
+#include <tesseract_monitoring/current_state_monitor.h>
+
+// Test run commands:
+// rosrun tesseract_monitoring robot_communication_monitor_node
+// rostopic pub -r 1 /joint_states sensor_msgs/JointState
+// rostopic echo /fanuc_com_error
+
+class FanucCommError
+{
+public:
+  FanucCommError(ros::NodeHandle& nh)
+  {
+    fanuc_comm_error.data = false;
+    chatter_pub = nh.advertise<std_msgs::Bool>("fanuc_com_error", 20);
```

Comment:
I would make this pull the topic name from a rosparam (or at least rename it). Also a lot of times hardcoded defaults like this will be defined at the top of the file so they are easy to find. e.g. at the top: `const std::string COM_ERROR_PUB_TOPIC_DEFAULT` or similar.

Then here `nh_.param<std::string>("error_topic", topic_name, COM_ERROR_PUB_TOPIC_DEFAULT)`

Reply:
I know this is unsollicited advice, so take it or leave it, but: don't configure topic names through parameters.

Hard-code some sensible name and then, if a consumer requires it, `remap` it to something else.